### ISSUE TITLE
Add support for parsing ORDINALITY column in JSON_TABLE for Oracle dialect

### ISSUE THREAD

**Issue by piatekmarcin** [sqlfluff/sqlfluff#7550]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Parsing of ORDINALITY column in JSON_TABLE, which is described in [Oracle Docs](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/[JSON_TABLE](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/JSON_TABLE.html#GUID-3C8E63B5-0B94-4E86-A2D3-3D4831B67C62__CJAHEFGD).html#GUID-3C8E63B5-0B94-4E86-A2D3-3D4831B67C62__CJAHEFGD), fails with parsing error.

```shell
==== parsing violations ====
L:   8 | P:  13 |  PRS | Line 8, Position 13: Found unparsable section: 'for ordinality,\n
                       | letter varchar(1...'
``` 

Example SQL statement containing ORDINALITY column:

```sql
select
    jt.seq,
    jt.letter
from json_table(
    '["A","B","C"]',
    '$[*]'
    columns (
        seq for ordinality,
        letter varchar(10) path '$'
    )
) as jt
``` 

### Expected Behaviour

No parsing error.

### Observed Behaviour

```shell
sqlfluff parse --dialect oracle test_json.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'select'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                newline:                                      '\n'
[L:  2, P:  1]      |                whitespace:                                   '    '
[L:  2, P:  5]      |                select_clause_element:
[L:  2, P:  5]      |                    column_reference:
[L:  2, P:  5]      |                        naked_identifier:                     'jt'
[L:  2, P:  7]      |                        dot:                                  '.'
[L:  2, P:  8]      |                        naked_identifier:                     'seq'
[L:  2, P: 11]      |                comma:                                        ','
[L:  2, P: 12]      |                newline:                                      '\n'
[L:  3, P:  1]      |                whitespace:                                   '    '
[L:  3, P:  5]      |                select_clause_element:
[L:  3, P:  5]      |                    column_reference:
[L:  3, P:  5]      |                        naked_identifier:                     'jt'
[L:  3, P:  7]      |                        dot:                                  '.'
[L:  3, P:  8]      |                        naked_identifier:                     'letter'
[L:  3, P: 14]      |                [META] dedent:
[L:  3, P: 14]      |            newline:                                          '\n'
[L:  4, P:  1]      |            from_clause:
[L:  4, P:  1]      |                keyword:                                      'from'
[L:  4, P:  5]      |                whitespace:                                   ' '
[L:  4, P:  6]      |                from_expression:
[L:  4, P:  6]      |                    [META] indent:
[L:  4, P:  6]      |                    from_expression_element:
[L:  4, P:  6]      |                        table_expression:
[L:  4, P:  6]      |                            function:
[L:  4, P:  6]      |                                function_name:
[L:  4, P:  6]      |                                    function_name_identifier:  'json_table'
[L:  4, P: 16]      |                                function_contents:
[L:  4, P: 16]      |                                    bracketed:
[L:  4, P: 16]      |                                        start_bracket:        '('
[L:  4, P: 17]      |                                        [META] indent:
[L:  4, P: 17]      |                                        newline:              '\n'
[L:  5, P:  1]      |                                        whitespace:           '    '
[L:  5, P:  5]      |                                        expression:
[L:  5, P:  5]      |                                            quoted_literal:   '\'["A","B","C"]\''
[L:  5, P: 20]      |                                        comma:                ','
[L:  5, P: 21]      |                                        newline:              '\n'
[L:  6, P:  1]      |                                        whitespace:           '    '
[L:  6, P:  5]      |                                        expression:
[L:  6, P:  5]      |                                            quoted_literal:   "'$[*]'"
[L:  6, P: 11]      |                                        newline:              '\n'
[L:  7, P:  1]      |                                        whitespace:           '    '
[L:  7, P:  5]      |                                        expression:
[L:  7, P:  5]      |                                            column_reference:
[L:  7, P:  5]      |                                                naked_identifier:  'columns'
[L:  7, P: 12]      |                                        whitespace:           ' '
[L:  7, P: 13]      |                                        expression:
[L:  7, P: 13]      |                                            bracketed:
[L:  7, P: 13]      |                                                start_bracket:  '('
[L:  7, P: 14]      |                                                [META] indent:
[L:  7, P: 14]      |                                                newline:      '\n'
[L:  8, P:  1]      |                                                whitespace:   '        '
[L:  8, P:  9]      |                                                expression:
[L:  8, P:  9]      |                                                    column_reference:
[L:  8, P:  9]      |                                                        naked_identifier:  'seq'
[L:  8, P: 12]      |                                                whitespace:   ' '
[L:  8, P: 13]      |                                                unparsable:   !! Expected: 'Nothing here.'
[L:  8, P: 13]      |                                                    word:     'for'
[L:  8, P: 16]      |                                                    whitespace:  ' '
[L:  8, P: 17]      |                                                    word:     'ordinality'
[L:  8, P: 27]      |                                                    comma:    ','
[L:  8, P: 28]      |                                                    newline:  '\n'
[L:  9, P:  1]      |                                                    whitespace:  '        '
[L:  9, P:  9]      |                                                    word:     'letter'
[L:  9, P: 15]      |                                                    whitespace:  ' '
[L:  9, P: 16]      |                                                    word:     'varchar'
[L:  9, P: 23]      |                                                    start_bracket:  '('
[L:  9, P: 24]      |                                                    numeric_literal:  '10'
[L:  9, P: 26]      |                                                    end_bracket:  ')'
[L:  9, P: 27]      |                                                    whitespace:  ' '
[L:  9, P: 28]      |                                                    word:     'path'
[L:  9, P: 32]      |                                                    whitespace:  ' '
[L:  9, P: 33]      |                                                    single_quote:  "'$'"
[L:  9, P: 36]      |                                                newline:      '\n'
[L: 10, P:  1]      |                                                whitespace:   '    '
[L: 10, P:  5]      |                                                [META] dedent:
[L: 10, P:  5]      |                                                end_bracket:  ')'
[L: 10, P:  6]      |                                        newline:              '\n'
[L: 11, P:  1]      |                                        [META] dedent:
[L: 11, P:  1]      |                                        end_bracket:          ')'
[L: 11, P:  2]      |                        whitespace:                           ' '
[L: 11, P:  3]      |                        alias_expression:
[L: 11, P:  3]      |                            [META] indent:
[L: 11, P:  3]      |                            alias_operator:
[L: 11, P:  3]      |                                keyword:                      'as'
[L: 11, P:  5]      |                            whitespace:                       ' '
[L: 11, P:  6]      |                            naked_identifier:                 'jt'
[L: 11, P:  8]      |                            [META] dedent:
[L: 11, P:  8]      |                    [META] dedent:
[L: 11, P:  8]      |    newline:                                                  '\n'
[L: 12, P:  1]      |    newline:                                                  '\n'
[L: 13, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   8 | P:  13 |  PRS | Line 8, Position 13: Found unparsable section: 'for ordinality,\n
                       | letter varchar(1...'
WARNING: Parsing errors found and dialect is set to 'oracle'. Have you configured your dialect correctly?

``` 

### How to reproduce

Save example SQL statement to `test_json.sql` file and execute command `sqlfluff parse --dialect oracle test_json.sql`

### Dialect

Oracle

### Version

```shell
==== sqlfluff ====
sqlfluff:                4.0.4 python:                 3.9.25
implementation:        cpython verbosity:                   1
dialect:                oracle templater:               jinja
rules:                                all

``` 

### Configuration

Default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
